### ISSUE TITLE
[terminal completions]: broken in `bash`/`zsh` on Unix /WSL

### ISSUE THREAD

**Issue by legomushroom** [microsoft/vscode#235331]
Follow up on: https://github.com/microsoft/vscode/issues/235021

The completions seem to be broken on Unix machines, in `bash`/`zsh` terminals. Please see https://github.com/microsoft/vscode/issues/235021#issuecomment-2515627451 for more info.

There's an error with `unable to resolve nonexistent file \user\home\` because it's using the wrong path separators.

**Comment by meganrogge** [comment]
Broken how? Do they ever show up?

**Comment by Tyriar** [comment]
@meganrogge I think they just didn't work at all, see https://github.com/microsoft/vscode/issues/235021#issuecomment-2515627451

**Comment by meganrogge** [comment]
This will be hard to investigate because will need to use a vm

**Comment by meganrogge** [comment]
Can reproduce on windows using WSL, nothing works at all

**Comment by meganrogge** [comment]
Interestingly, it doesn't show the `No suggestions` widget either, though I see the command is correctly being invoked. It's like the widget doesn't get created 🤔

**Comment by meganrogge** [comment]
wondering if maybe this will fix it https://github.com/microsoft/vscode/pull/236630

**Comment by Tyriar** [comment]
Working for me on 1.98 insiders:

![Image](https://github.com/user-attachments/assets/dcc110fa-12fc-4e3c-b5ed-1795bab672a3)

![Image](https://github.com/user-attachments/assets/13aca916-7f10-4133-99a8-70df3507ccc7)

**Comment by rzhao271** [comment]
Not working in WSL; looks like it's trying to use backslashes in the path.

![Completions in WSL result in folder not found error](https://github.com/user-attachments/assets/374466f0-5aac-48e6-8408-165bf4478270)

**Comment by meganrogge** [comment]
I think we need to be using `isWindowsPty` here vs `isWindows`

**Comment by argentinaluiz** [comment]
Me too with WSL (bash and zsh)

![Image](https://github.com/user-attachments/assets/651a5631-6ace-436d-8df7-6f4a435ea32d)

![Image](https://github.com/user-attachments/assets/abf321b8-9f56-4fe8-a010-5d1094079a53)

**Comment by meganrogge** [comment]
deep dive in dev tools shows we have the right path sep /root. i think the issue could be we have scheme file here and it's a dir ? but then why does it work when scheme is file for other platforms... 🤔 

more likely theory: file service changes the URI's path sep based on the local OS, which throws this off. It throws here.

the error is `unable to resolve nonexistent file \root`

![Image](https://github.com/user-attachments/assets/56f1614f-b256-43f6-b5c3-66f9362d6597)

**Comment by meganrogge** [comment]
@aeschli do you have any insights here?

**Comment by aeschli** [comment]
Yes, something with the URI tarnsformation is not working here, the code in core should see `vscode-remote/wsl+ubuntu/...`
See https://github.com/microsoft/vscode/pull/251129#issuecomment-2965881260

Ideally all the file system accesses are done on the extension side. So, not sure why core is accessing the file system here.

**Comment by meganrogge** [comment]
We decided to keep the file resolving in core vs the extension so that's handled uniformly vs up to terminal completion providers.

**Comment by Tyriar** [comment]
@aeschli many extensions can provide files, but it's not a list but a request of what to show. Those requests are collected and consolidated on the main side so we only do this work once. For remote scenarios that does mean we need to go back to the remote to resolve things in some cases.

@meganrogge I still see this in today's build:

![Image](https://github.com/user-attachments/assets/04afade5-135a-4436-8f54-64ff5be13fdd)

**Comment by Tyriar** [comment]
The scheme is wrong (debugging latest insiders):

![Image](https://github.com/user-attachments/assets/00bc459b-de63-4300-b8ab-ee24968b874e)

https://github.com/microsoft/vscode/blob/88678d8a1dd80744ee65c581836dd068239d3ef5/src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalCompletionService.ts#L219

**Comment by meganrogge** [comment]
@Tyriar don't think that has the fix

**Comment by connor4312** [comment]
This did not work for me on bash in WSL. Triggering completions with ctrl+space resulted in a stuck "Loading..." state.

Trace logs from console:

<img width="1269" height="133" alt="Image" src="https://github.com/user-attachments/assets/3d03d2c9-7316-4097-9bad-e94184e2407a" />

Terminal output channel:

[log (3).txt](https://github.com/user-attachments/files/21041651/log.3.txt)

**Comment by Tyriar** [comment]
WSL/bash for me :

<img width="534" height="295" alt="Image" src="https://github.com/user-attachments/assets/74c3e5c7-d3fd-46f5-95cb-1b6f1f0af46e" />

Files (slash problem in https://github.com/microsoft/vscode/issues/253970):

<img width="537" height="286" alt="Image" src="https://github.com/user-attachments/assets/6a59d6d7-27bc-4ad7-baaf-e4c5b50cb602" />

**Comment by Tyriar** [comment]
From Connor's log, the prompt input model looks correct:

```
2025-07-03 09:44:54.941 [debug] [512b81c] parsing data "/" [[47]]
2025-07-03 09:44:54.941 [trace] [512b81c] PromptInputModel#_sync: ls .| []
2025-07-03 09:44:54.942 [trace] [512b81c] PromptInputModel#onDidChangeInput ["ls ./|"]
2025-07-03 09:44:54.943 [trace] [512b81c] PromptInputModel#_sync: ls ./| []
```

@meganrogge I think we might be lacking log statements in important renderer-side methods like `showSuggestions` and `acceptSelectedSuggestion`.

**Comment by meganrogge** [comment]
@rzhao271 also saw this for `bash`

**Comment by meganrogge** [comment]
@rzhao271, we added log statements, would you mind checking them out? https://github.com/microsoft/vscode/blob/ff478bd465579f50e6089f3e93adda6b207f2856/src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalCompletionService.ts#L182-L189

I expect you might see the timeout happening

**Comment by rzhao271** [comment]
Ironically, the feature LGTM now. I pressed Ctrl+Space in a terminal on the latest Insiders in Ubuntu WSL. It took a while for the suggestions to show up, but they did show up this time.

<img width="1000" height="319" alt="Paths represented properly after triggering suggestion" src="https://github.com/user-attachments/assets/1a800819-e29e-4a29-b9e3-0b377019c939" />
